### PR TITLE
Add activity plot function

### PR DESCRIPTION
**Pull request by thequilo**
This is a rough draft of the function. There are many things that could need discussion or improvement:
 - naming of the functions and their parameters
 - How to pass through kwargs to lower-level functions
 - Is a `dict` of `ArrayIntervals` the correct representation? Would it be easier to also support lists of tuples of start and end times? or a tuple of two lists of start and end times? The raw json contains a list of start times and a list of end times
 - Generalize hbar even more so that other args like color are broadcasted to the intervals in the `ArrayInterval`s?
 - Is this the correct location for the `activity` function or is it too specific? Would it be better located elsewhere, e.g., next to the database?

### CODE REVIEW

File: paderbox/visualization/plot.py [hunk 1]
```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
...
+    Args:
+        y: The `y` coordinate(s). Can be a scalar (all bars are drawn at the
+            same y coordinate) or array-like of length `len(width)` (each
+            element of `width` has its own y coordinate)
+        width: Scalar, array-like, `pb.array.interval.ArrayInterval` or list of
+            `pb.array.interval.ArrayInterval`.
+        height:
+        left:
+        ax:
+        scale:
```
Comment (jensheit) [inline paderbox/visualization/plot.py:873]: should we copy the args definition from plt.barh?

File: paderbox/visualization/plot.py [hunk 2]
```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
...
+        width = new_width
+        y = new_y
+    elif not isinstance(width, Iterable):
+        width = [width]
+    ax.barh(y, width, height, left, **kwargs)
+    return ax
+
+
+@create_subplot
+def axvspan(start, stop=None, ax=None, **kwargs):
```
Comment (jensheit) [inline paderbox/visualization/plot.py:922]: this code does not work with the default ax, so either we have to change the default or reset it to plt if ax=None

File: paderbox/visualization/plot.py [hunk 3]
```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
...
+        activity_intervals:
+        segment_boundary_intervals:
+        ax:
+        activity_kwargs:
+        overlap_kwargs:
+        total_activity_kwargs:
+        scale:
+        bar_edgecolor:
+        plot_total_activity:
+
```
Comment (jensheit) [inline paderbox/visualization/plot.py:973]: I think you need to add some explanation here, at least for activity_intervals and segment_boundary_intervals

Comment (jensheit) [reply paderbox/visualization/plot.py:973]: y_labels is missing

File: paderbox/visualization/plot.py [hunk 4]
```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
...
+    Returns:
+
+    """
+    import numpy as np
+    import paderbox as pb
```
Comment (jensheit) [inline paderbox/visualization/plot.py:978]: do we really need to import those to inside of a function?

File: paderbox/visualization/plot.py [hunk 5]
```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
...
+    ))
+    num_active = np.zeros((length,), dtype=np.int)
+    for a in activity_intervals:
+        num_active += a[:length]
+
+    # Plot overlaps
+    if plot_overlaps:
+        kwargs = {'color': 'orange', 'alpha': 0.4}
+        if overlap_kwargs is not None:
+            kwargs.update(overlap_kwargs)
```
Comment (jensheit) [inline paderbox/visualization/plot.py:1004]: sollte man hier wirklich die standard kwargs verändern?

File: paderbox/visualization/plot.py [hunk 6]
```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
...
+@create_subplot
+def activity(
+        activity_intervals,
+        y_labels=None,
+        segment_boundary_intervals=None,
+        ax=None,
+        activity_kwargs=None,
+        overlap_kwargs=None,
+        total_activity_kwargs=None,
+        plot_total_activity=True,
```
Comment (jensheit) [inline paderbox/visualization/plot.py:956]: I would move this option in front of the kwargs
Am I right to assume, that this plots activity bar if any speaker is active?

File: paderbox/visualization/plot.py (hunk 1 above)
Comment (thequilo) [reply paderbox/visualization/plot.py:873]: Yes, we could do that. Or write "see plt.barh".

File: paderbox/visualization/plot.py (hunk 2 above)
Comment (thequilo) [reply paderbox/visualization/plot.py:922]: It works with the default `ax`. It gets filled by the `create_subplot` decorator, as for all other functions in this file

File: paderbox/visualization/plot.py (hunk 4 above)
Comment (thequilo) [reply paderbox/visualization/plot.py:978]: No, in my private repository, where I copied the function from, this was the only function that used np and pb, but here other functions also use np.

File: paderbox/visualization/plot.py (hunk 6 above)
Comment (thequilo) [reply paderbox/visualization/plot.py:956]: `plot_total_acitivty` plots an additional bar for the overall activity (green bar in the image)
image

**Comment by thequilo** [comment]
Well, something went wrong with the rebase on master

File: paderbox/visualization/plot.py (hunk 2 above)
Comment (jensheit) [reply paderbox/visualization/plot.py:919]: ah I didnt see the decorator you are right of course

**Review by jensheit — APPROVED** [review]

**Pull request merged**